Installation issue: suite-sparse 5.3.0

### Steps to reproduce the issue

```console
$ spack install suite-sparse
==> pkgconf is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/pkgconf-1.5.4-govrjwe257pbvbo3uuv2zlgjb7qbchiv
==> ncurses is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/ncurses-6.1-7m7kwj3etap7iwagymy5ti3fen3jizh7
==> readline is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/readline-7.0-q2qwhp7vn6mgg7kxoqw6semcx7jiwhhf
==> gdbm is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/gdbm-1.18.1-qmpp2pk4aseqoupn4c2lwyozhq4clqui
==> perl is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/perl-5.26.2-za57evsnti7bgysyttea6lb2zbvmgvgd
==> zlib is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/zlib-1.2.11-lgevn3ckxbm4sxwjdoq3k5qfgmpsjt7o
==> openssl is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/openssl-1.1.1-ptmpqg6qpzbiqigrgw54nk2ecm7h7o5y
==> cmake is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/cmake-3.13.2-6zy5lexck2yepsfv2ju4ujjczpnbkx65
==> metis is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/metis-5.1.0-jj2hpmf3df6v37rsc77lec26ep7ac5bj
==> openblas is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/openblas-0.3.4-civzquuc27y2ar4g2oaqtu7b3gwkjvgz
==> Installing suite-sparse
==> Searching for binary cache of suite-sparse
==> Warning: No Spack mirrors are currently configured
==> No binary for suite-sparse found: installing from source
==> Using cached archive: /home/ben/Downloads/spack-cache/suite-sparse/suite-sparse-5.3.0.tar.gz
==> Staging archive: /home/ben/projects/spack/spack/var/spack/stage/suite-sparse-5.3.0-xd4moigqqgtoqzhs3xw53wk3lal2ayr6/SuiteSparse-5.3.0.tar.gz
==> Created stage in /home/ben/projects/spack/spack/var/spack/stage/suite-sparse-5.3.0-xd4moigqqgtoqzhs3xw53wk3lal2ayr6
==> No patches needed for suite-sparse
==> Building suite-sparse [Package]
==> Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j8' 'install' 'INSTALL=/opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/suite-sparse-5.3.0-xd4moigqqgtoqzhs3xw53wk3lal2ayr6' 'AUTOCC=no' 'CUDA=no' 'CUDA_PATH=' 'CFOPENMP=' 'CFLAGS=-O3 -fPIC' 'FFLAGS=-fPIC' 'F77FLAGS=-fPIC' 'MY_METIS_LIB=-L/opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/metis-5.1.0-jj2hpmf3df6v37rsc77lec26ep7ac5bj/lib -lmetis' 'MY_METIS_INC=/opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/metis-5.1.0-jj2hpmf3df6v37rsc77lec26ep7ac5bj/include' 'BLAS=-L/opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/openblas-0.3.4-civzquuc27y2ar4g2oaqtu7b3gwkjvgz/lib -lopenblas' 'LAPACK=-L/opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/openblas-0.3.4-civzquuc27y2ar4g2oaqtu7b3gwkjvgz/lib -lopenblas' 'CFLAGS+=-fno-common -fexceptions'

12 errors found in build log:
     2718    [ 92%] Building CXX object CMakeFiles/mongoose_exe.dir/Executable/mongoose.o
     2719    [ 94%] Building CXX object CMakeFiles/mongoose_test_performance.dir/Tests/Mongoose_Test_Performance_exe.o
     2720    [ 94%] Building CXX object CMakeFiles/mongoose_test_performance.dir/Tests/Mongoose_Test_Performance.o
     2721    [ 95%] Building CXX object CMakeFiles/mongoose_test_reference.dir/Tests/Mongoose_Test_Reference.o
     2722    [ 95%] Building CXX object CMakeFiles/mongoose_test_reference.dir/Tests/Mongoose_Test_Reference_exe.o
     2723    /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/6/../../../x86_64-linux-gnu/libsuitesparseconfig.a(SuiteSparse_config.o): relocation R_X86_64_PC32 against symbol `SuiteSparse_config' can not be used when making a shared object; re
             compile with -fPIC
  >> 2724    /usr/bin/ld: final link failed: Bad value
  >> 2725    collect2: error: ld returned 1 exit status
     2726    CMakeFiles/mongoose_dylib.dir/build.make:504: recipe for target 'lib/libmongoose.so.2.0.2' failed
  >> 2727    make[4]: *** [lib/libmongoose.so.2.0.2] Error 1
     2728    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2729    CMakeFiles/Makefile2:321: recipe for target 'CMakeFiles/mongoose_dylib.dir/all' failed
  >> 2730    make[3]: *** [CMakeFiles/mongoose_dylib.dir/all] Error 2
     2731    make[3]: *** Waiting for unfinished jobs....
     2732    [ 96%] Linking CXX executable tests/mongoose_test_memory
     2733    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2734    [ 96%] Built target mongoose_test_memory
     2735    [ 97%] Linking CXX executable tests/mongoose_test_performance
     2736    [ 98%] Linking CXX executable tests/mongoose_test_reference

     ...

     2743    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2744    [100%] Built target demo_exe
     2745    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2746    [100%] Built target mongoose_exe
     2747    make[3]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2748    Makefile:140: recipe for target 'all' failed
  >> 2749    make[2]: *** [all] Error 2
     2750    make[2]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2751    make[2]: Entering directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2752    make[3]: Entering directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2753    make[4]: Entering directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2754    make[4]: Entering directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2755    make[4]: Entering directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'

     ...

     2788    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2789    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2790    [ 69%] Built target mongoose_test_memory
     2791    [ 72%] Built target mongoose_test_edgesep
     2792    [ 75%] Built target mongoose_test_performance
     2793    /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/6/../../../x86_64-linux-gnu/libsuitesparseconfig.a(SuiteSparse_config.o): relocation R_X86_64_PC32 against symbol `SuiteSparse_config' can not be used when making a shared object; re
             compile with -fPIC
  >> 2794    /usr/bin/ld: final link failed: Bad value
  >> 2795    collect2: error: ld returned 1 exit status
     2796    CMakeFiles/mongoose_dylib.dir/build.make:504: recipe for target 'lib/libmongoose.so.2.0.2' failed
  >> 2797    make[4]: *** [lib/libmongoose.so.2.0.2] Error 1
     2798    make[4]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2799    CMakeFiles/Makefile2:321: recipe for target 'CMakeFiles/mongoose_dylib.dir/all' failed
  >> 2800    make[3]: *** [CMakeFiles/mongoose_dylib.dir/all] Error 2
     2801    make[3]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2802    Makefile:140: recipe for target 'all' failed
  >> 2803    make[2]: *** [all] Error 2
     2804    make[2]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose/build'
     2805    Makefile:28: recipe for target 'install' failed
  >> 2806    make[1]: *** [install] Error 2
     2807    make[1]: Leaving directory '/tmp/ben/spack-stage/spack-stage-DwNOha/SuiteSparse/Mongoose'
     2808    Makefile:40: recipe for target 'install' failed
  >> 2809    make: *** [install] Error 2

See build log for details:
  /home/ben/projects/spack/spack/var/spack/stage/suite-sparse-5.3.0-xd4moigqqgtoqzhs3xw53wk3lal2ayr6/SuiteSparse/spack-build.out
```

### Platform and user environment

Please report your OS here:
```console
$ uname -a
Linux 4.9.0-8-amd64 #1 SMP Debian 4.9.130-2 (2018-10-27) x86_64 GNU/Linux
x86_64 GNU/Linux
$ lsb_release -d
Description:    Debian GNU/Linux 9.6 (stretch)
``` 
I have the following Debian packages installed:
```console
$ aptitude search suitesparse | grep ^i
i  libsuitesparse-dev - libraries for sparse matrices computations (development files)
i A libsuitesparseconfig4 - configuration routines for all SuiteSparse modules
```

### Additional information

```console
$ spack spec --install-status suite-sparse
Input spec
--------------------------------
 -   suite-sparse

Concretized
--------------------------------
 -   suite-sparse@5.3.0%gcc@6.3.0~cuda~openmp+pic~tbb arch=linux-debian9-x86_64 
[+]      ^cmake@3.13.2%gcc@6.3.0~doc+ncurses+openssl+ownlibs~qt arch=linux-debian9-x86_64 
[+]          ^ncurses@6.1%gcc@6.3.0~symlinks~termlib arch=linux-debian9-x86_64 
[+]              ^pkgconf@1.5.4%gcc@6.3.0 arch=linux-debian9-x86_64 
[+]          ^openssl@1.1.1%gcc@6.3.0+systemcerts arch=linux-debian9-x86_64 
[+]              ^perl@5.26.2%gcc@6.3.0+cpanm patches=0eac10ed90aeb0459ad8851f88081d439a4e41978e586ec743069e8b059370ac +shared+threads arch=linux-debian9-x86_64 
[+]                  ^gdbm@1.18.1%gcc@6.3.0 arch=linux-debian9-x86_64 
[+]                      ^readline@7.0%gcc@6.3.0 arch=linux-debian9-x86_64 
[+]              ^zlib@1.2.11%gcc@6.3.0+optimize+pic+shared arch=linux-debian9-x86_64 
[+]      ^metis@5.1.0%gcc@6.3.0 build_type=Release ~gdb~int64 patches=4991da938c1d3a1d3dea78e49bbebecba00273f98df2a656e38b83d55b281da1,b1225da886605ea558db7ac08dd8054742ea5afe5ed61ad4d0fe7a495b1270d2 ~real64+shared arch=linux-debian9-x86_64 
[+]      ^openblas@0.3.4%gcc@6.3.0 cpu_target= ~ilp64+pic+shared threads=none ~virtual_machine arch=linux-debian9-x86_64 
```

The previous version can be installed with:
```console
$ spack install suite-sparse@5.2.0
==> suite-sparse is already installed in /opt/spack/opt/linux-debian9-x86_64/gcc-6.3.0/suite-sparse-5.2.0-ch2ebcppbbafra3h3symyhjfryyxid3q
```

--- a/var/spack/repos/builtin/packages/suite-sparse/package.py
+++ b/var/spack/repos/builtin/packages/suite-sparse/package.py
@@ -34,6 +34,7 @@ class SuiteSparse(Package):
 
     depends_on('blas')
     depends_on('lapack')
+    depends_on('m4', type='build', when='@5.0.0:')
     depends_on('cmake', when='@5.2.0:', type='build')
 
     depends_on('metis@5.1.0', when='@4.5.1:')
@@ -64,7 +65,6 @@ class SuiteSparse(Package):
         pic_flag  = self.compiler.pic_flag if '+pic' in spec else ''
 
         make_args = [
-            'INSTALL=%s' % prefix,
             # By default, the Makefile uses the Intel compilers if
             # they are found. The AUTOCC flag disables this behavior,
             # forcing it to use Spack's compiler wrappers.
@@ -134,6 +134,7 @@ class SuiteSparse(Package):
         if '@5.4.0:' in self.spec:
             make('default', *make_args)
 
+        make_args.append('INSTALL=%s' % prefix)
         make('install', *make_args)
 
     @property

